cooltools/cli/call_dots.py

```diff
@@ -197,60 +212,115 @@ def call_dots(
-    'chrom', 'diag', 'n_valid', value_name. value_name is controlled using
+    'region', 'diag', 'n_valid', value_name. value_name is controlled using
     options. Header must be present in a file.
 
     """
     clr = cooler.Cooler(cool_path)
 
-    expected_columns = ["chrom", "diag", "n_valid", expected_name]
-    expected_index = ["chrom", "diag"]
+    # preliminary SCHEMA for cis-expected
+    region_column_name = "region"
+    expected_columns = [region_column_name, "diag", "n_valid", expected_name]
     expected_dtypes = {
-        "chrom": np.str,
+        region_column_name: np.str,
         "diag": np.int64,
         "n_valid": np.int64,
         expected_name: np.float64,
     }
-    expected = pd.read_table(
-        expected_path,
-        usecols=expected_columns,
-        dtype=expected_dtypes,
-        comment=None,
-        verbose=verbose,
-    )
+
+    try:
+        expected = pd.read_table(
+            expected_path,
+            usecols=expected_columns,
+            dtype=expected_dtypes,
+            comment=None,
+            verbose=verbose,
+        )
+    except ValueError as e:
+        raise ValueError(
+            "input expected does not match the schema\n"
+            "tab-separated expected file must have a header as wel"
+        )
+    expected_index = [
+        region_column_name,
+        "diag",
+    ]
     expected.set_index(expected_index, inplace=True)
+    # end of SCHEMA for cis-expected
+
+    # Optional reading region table provided by the user:
+    if regions is None:
+        try:
+            uniq_regions = expected.index.get_level_values(region_column_name).unique()
+            regions_table = bioframe.parse_regions(uniq_regions, clr.chromsizes)
+            regions_table["name"] = regions_table["chrom"]
+        except ValueError as e:
+            print(e)
+            raise ValueError(
+                "Cannot interpret regions from EXPECTED_PATH\n"
+                "specify regions definitions using --regions option."
+            )
+    else:
+        # Flexible reading of the regions table:
+        regions_buf, names = util.sniff_for_header(regions)
+        regions_table = pd.read_csv(regions_buf, sep="\t", header=None)
+        if regions_table.shape[1] not in (3, 4):
+            raise ValueError(
+                "The region file does not have three or four tab-delimited columns."
+                "We expect a bed file with columns chrom, start, end, and optional name"
+            )
+        if regions_table.shape[1] == 4:
+            regions_table = regions_table.rename(
+                columns={0: "chrom", 1: "start", 2: "end", 3: "name"}
+            )
+            regions_table = bioframe.parse_regions(regions_table)
+        else:
+            regions_table = regions_table.rename(
+                columns={0: "chrom", 1: "start", 2: "end"}
+            )
+            regions_table["name"] = list(
```

Comment:
@agalitsyna I think `bioframe.parse_regions` does that automatically ... - if you give it a 3 column bed - i'll auto-generate UCSC names for it.

So I wonder if this whole `if regions_table.shape[1] == 4` is really needed - I think `bioframe.parse_regions` could handle both 3 and 4 columns BED-like dataframes - shall we try that instead ?

Reply:
3,4 - are needed as it turned out

Reply:
I removed `name` assignment - otherwise intact